File: skill_sdk/tests/fixtures/cpp-project/src/shop/InMemoryRepository.cpp
#include "InMemoryRepository.h"

InMemoryRepository::InMemoryRepository() {}

bool InMemoryRepository::SaveOrder(const Order& order) {
    orders_[order.OrderID] = order;
    return true;
}

Order InMemoryRepository::GetOrder(const std::string& orderID) {
    auto it = orders_.find(orderID);
    return it != orders_.end() ? it->second : Order{};
}

bool InMemoryRepository::UpdateOrderStatus(const std::string& orderID, OrderStatus status) {
    auto it = orders_.find(orderID);
    if (it == orders_.end()) return false;
    it->second.Status = status;
    return true;
}

Product InMemoryRepository::GetProduct(const std::string& sku) {
    auto it = products_.find(sku);
    return it != products_.end() ? it->second : Product{};
}

std::vector<Order> InMemoryRepository::ListOrdersByUser(const std::string& userID) {
    std::vector<Order> result;
    for (const auto& [_, order] : orders_) {
        if (order.UserID == userID)
            result.push_back(order);
    }
    return result;
}

File: skill_sdk/tests/fixtures/js-project/src/shop/repository/InMemoryRepository.js
import { Repository } from './Repository.js';

/**
 * In-memory implementation of Repository.
 *
 * @implements {Repository}
 */
export class InMemoryRepository extends Repository {
  constructor() {
    super();
    /** @type {Map<string, import('../models/Order.js').Order>} */
    this._orders = new Map();
    /** @type {Map<string, import('../models/Product.js').Product>} */
    this._products = new Map();
  }

  /** @param {import('../models/Order.js').Order} order */
  saveOrder(order) {
    this._orders.set(order.getOrderId(), order);
  }

  /**
   * @param {string} orderId
   * @returns {import('../models/Order.js').Order | undefined}
   */
  getOrder(orderId) {
    return this._orders.get(orderId);
  }

  /**
   * @param {string} orderId
   * @param {string} status
   */
  updateOrderStatus(orderId, status) {
    const order = this._orders.get(orderId);
    if (order) {
      order.setStatus(status);
    }
  }

  /**
   * @param {string} sku
   * @returns {import('../models/Product.js').Product | undefined}
   */
  getProduct(sku) {
    return this._products.get(sku);
  }

  /**
   * @param {string} userId
   * @returns {import('../models/Order.js').Order[]}
   */
  listOrdersByUser(userId) {
    const result = [];
    for (const order of this._orders.values()) {
      if (order.getUserId() === userId) {
        result.push(order);
      }
    }
    return result;
  }
}
